packages/grafana-ui/src/components/VizLegend/VizLegend.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { FC, CSSProperties } from 'react';
+import { VizOrientation } from '@grafana/data';
+
+interface Props {
+  orientation: VizOrientation;
+  children: React.ReactNode;
+}
+
+export const VizLegend: FC<Props> = ({ children, orientation }) => {
+  const styles: CSSProperties = {
+    display: 'flex',
+  };
+
+  if (orientation === VizOrientation.Vertical) {
+    styles.flexDirection = 'column';
+  }
+
+  return <div style={styles}>{children}</div>;
+};
+
+export interface VizLegendItemProps {
+  color: string;
+  name: string;
+  value?: string;
```

Comment:
could just pass in a DisplayValue here, but was thinking of making this component display only and not care much about data types